src/firestore/path.ts

```diff
@@ -0,0 +1,90 @@
+import * as FirebaseError from "../error";
+
+export class FirestorePath {
+  path: string;
+  parent: string;
+
+  constructor(project: string, path: string) {
+    this.path = path;
+
+    // Remove any leading or trailing slashes from the path
+    if (this.path) {
+      this.path = path.replace(/(^\/+|\/+$)/g, "");
+    }
+
+    this.parent = "projects/" + project + "/databases/(default)/documents";
+  }
+
+  validate(): void {
+    const pieces = this.path.split("/");
+
+    if (pieces.length === 0) {
+      throw new FirebaseError("Path length must be greater than zero.");
+    }
+
+    const hasEmptySegment = pieces.some((piece) => {
+      return piece.length === 0;
+    });
+
+    if (hasEmptySegment) {
+      throw new FirebaseError("Path must not have any empty segments.");
+    }
+  }
+
+  isDocumentPath(): boolean {
+    return this._isDocumentPath(this.path);
```

Comment:
I think, since this is only a one-liner, having the private function `_isDocumentPath` is unnecessary